src/main/java/com/milaboratory/mixcr/basictypes/AlignmentsIO.java

```diff
@@ -0,0 +1,325 @@
+ * three paragraphs appear in all copies.
+ *
+ * Those desiring to incorporate this work into commercial products or use for
+ * commercial purposes should contact MiLaboratory LLC, which owns exclusive
+ * rights for distribution of this program for commercial purposes, using the
+ * following email address: licensing@milaboratory.com.
+ *
+ * IN NO EVENT SHALL THE INVENTORS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT,
+ * SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS,
+ * ARISING OUT OF THE USE OF THIS SOFTWARE, EVEN IF THE INVENTORS HAS BEEN
+ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * THE SOFTWARE PROVIDED HEREIN IS ON AN "AS IS" BASIS, AND THE INVENTORS HAS
+ * NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR
+ * MODIFICATIONS. THE INVENTORS MAKES NO REPRESENTATIONS AND EXTENDS NO
+ * WARRANTIES OF ANY KIND, EITHER IMPLIED OR EXPRESS, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY OR FITNESS FOR A
+ * PARTICULAR PURPOSE, OR THAT THE USE OF THE SOFTWARE WILL NOT INFRINGE ANY
+ * PATENT, TRADEMARK OR OTHER RIGHTS.
+ */
+package com.milaboratory.mixcr.basictypes;
+
+import com.milaboratory.primitivio.PrimitivI;
+import com.milaboratory.primitivio.PrimitivIState;
+import com.milaboratory.primitivio.PrimitivO;
+import com.milaboratory.primitivio.PrimitivOState;
+import com.milaboratory.util.ByteArrayDataOutput;
+import com.milaboratory.util.ByteBufferDataInputAdapter;
+import net.jpountz.lz4.LZ4Compressor;
+import net.jpountz.lz4.LZ4FastDecompressor;
+import net.jpountz.xxhash.XXHash32;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Utility methods for block-compressed alignment objects IO.
+ *
+ * Block:
+ *
+ * Header (17 bytes total):
+ * [ 1 byte : bit0 = (0 = last block; 1 = data block); bit1 = (1 = compressed ; 0 = raw) ]
+ * [ 4 bytes : int : number of alignments ]
+ * [ 4 bytes : int : rawDataSize ]
+ * [ 4 bytes : int : compressedDataSize ]
+ * [ 4 bytes : int : checksum for raw data ]
+ *
+ * Data:
+ * [ dataSize bytes ] (compressed, if bit1 of header is 1; uncompressed, if bit1 is 0 )
+ */
+public final class AlignmentsIO {
+    public static final int HASH_SEED = 0xD5D20F71;
+    public static final byte LAST_BYTE = 0; // kB
+    public static final int AVERAGE_ALIGNMENT_SIZE = 1024; // kB
+
+    public static final int DEFAULT_ALIGNMENTS_IN_BLOCK = 1024; // 1024 alignments * 805-1024 bytes per alignment ~  824 kB - 1MB per block
+
+    private AlignmentsIO() {
+    }
+
+    static final int BLOCK_HEADER_SIZE = 17;
+
+    public static void writeIntBE(int val, byte[] buffer, int offset) {
+        buffer[offset++] = (byte) (val >>> 24);
```

Comment:
This will be faster:

```
buffer[offset] = (byte) (val >>> 24);
buffer[offset + 1] = (byte) (val >>> 16);
buffer[offset + 2] = (byte) (val >>> 8);
buffer[offset + 3] = (byte) val;
```
:)